sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -490,6 +491,56 @@ def test_spline_transformer_n_features_out(
     assert splt.transform(X).shape[1] == splt.n_features_out_
 
 
+@pytest.mark.parametrize("knots", ["uniform", "quantile"])
+@pytest.mark.parametrize(
+    "extrapolation", ["error", "constant", "linear", "continue", "periodic"]
+)
+@pytest.mark.parametrize("sparse_output", [False, True])
+def test_spline_transformer_handles_missing_values(knots, extrapolation, sparse_output):
+    """Test that SplineTransformer handles missing values correctly."""
+    X_nan = [[1, 1], [2, 2], [3, 3], [np.nan, 4], [4, 4]]
+    X = [[1, 1], [2, 2], [3, 3], [4, 4]]
+
+    # only for development, will later be removed
+    # X_nan = [[1], [2], [3], [np.nan], [4]]
+    # X = [[1], [2], [3], [4]]
+
+    # check correct error message for handle_missing="error"
+    msg = (
+        "'X' contains np.nan values, which is conflicting with handle_missing='error'."
+    )
+    with raises(ValueError, match=msg):
+        spline = SplineTransformer(
+            degree=2,
+            n_knots=3,
+            knots=knots,
+            handle_missing="error",
+            extrapolation=extrapolation,
+        )
+        spline.fit_transform(X_nan)
+
```

Comment:
when `handle_missing != "error"` please check that we have generic invariant that hold:

```python
X_nan_fit_transformed = spline.fit_transform(X_nan)
assert (X_nan_fit_transformed >= 0).all()
assert (X_nan_fit_transformed <= 1).all()

X_nan_fit_then_transformed = spline.fit(X_nan).transform(X_nan)
assert_allclose(X_nan_fit_transformed, X_nan_fit_then_transformed)

# Check that .transform works as expected when the passed data has not the same
# shape as the training set array:
assert_allclose(spline.fit_transform(X_nan)[::2], spline.fit(X_nan).transform(X_nan[::2]))  
```

When `handle_missing == "indicator"` we should also have:

```python
assert_allclose(X_nan_fit_transformed.sum(axis=1), np.ones(X_nan.shape[0]))
```

Reply:
I have added these tests, but I am unsure about the last one. 

Since with nans making zero-splines, those columns cannot add up to the same number as the others, which is expected.
Also, it seems like the number all the splines add up to scales by the number of features.

Do you think that I should adjust the test? Or is is rather that we should not change the properties/generic invariants of the transformer and thus the implementation of adding nan support should be different?